### ISSUE TITLE
JaxRSServerCodegen overrides JavaClientCodegen's getTypeDeclaration() method with an incorrect implementation

### ISSUE THREAD

**Issue by Helmsdown** [swagger-api/swagger-codegen#996]
So JavaClientCodegen has a good implementation of getTypeDeclaration() which correctly handles MapProperty objects.

```java
    public String getTypeDeclaration(Property p) {
        Property inner;
        if(p instanceof ArrayProperty) {
            ArrayProperty mp1 = (ArrayProperty)p;
            inner = mp1.getItems();
            return this.getSwaggerType(p) + "<" + this.getTypeDeclaration(inner) + ">";
        } else if(p instanceof MapProperty) {
            MapProperty mp = (MapProperty)p;
            inner = mp.getAdditionalProperties();
            return this.getSwaggerType(p) + "<String, " + this.getTypeDeclaration(inner) + ">";
        } else {
            return super.getTypeDeclaration(p);
        }
    }
```

JaxRSServerCodegen overrides this method with and drops some important code that handles MapProperty objects.

```java
    public String getTypeDeclaration(Property p) {
        Property inner;
        if(p instanceof ArrayProperty) {
            ArrayProperty mp1 = (ArrayProperty)p;
            inner = mp1.getItems();
            return this.getSwaggerType(p) + "<" + this.getTypeDeclaration(inner) + ">";
        } else if(p instanceof MapProperty) {
            MapProperty mp = (MapProperty)p;
            inner = mp.getAdditionalProperties();
            return this.getTypeDeclaration(inner);
        } else {
            return super.getTypeDeclaration(p);
        }
    }
```

The JavaClientCodegen implementation seems like it would work when generating the JaxRS template. 

Can we just get rid of the overriding method in JaxRSServerCodegen? WDYT?

**Comment by olensmar** [comment]
sounds reasonable to me - @fehguy - any objections?

**Comment by wing328** [comment]
@Helmsdown thanks for the suggestion. There's a similar PR submitted but for `SpringMVCServerCodegen`

**Comment by wing328** [comment]
PR for SpringMVCServerCodegen: https://github.com/swagger-api/swagger-codegen/pull/961

**Comment by Helmsdown** [comment]
Included a fix for this in #1067

**Comment by Helmsdown** [comment]
Fixed via #1067. Closing.
